File: 2022/25/java/Main.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.stream.Stream;

public class Main {
	private static final String INPUT_FILE = "../input/input.txt";

	public static void main(String[] args) throws IOException {
		try (Stream<String> stream = Files.lines(Paths.get(INPUT_FILE))) {
			long total = stream.mapToLong(line -> SNAFU.from(line)).sum();
			String snafu = SNAFU.to(total);
			System.out.format("SNAFU number to supply to Bob's console: %s%n", snafu);
		}
	}
}

File: 2022/25/javascript/main.mjs
import fs from "node:fs";
import readline from "node:readline";
import { from, to } from "./snafu.mjs";

// --- Constants ---
const INPUT_FILE = "../input/input.txt";

// --- Variables ---
let total = 0;

// --- Parse the input file ---
const stream = fs.createReadStream(INPUT_FILE);
const reader = readline.createInterface({ input: stream, crlDelay: Infinity });
reader.on("line", line => {
	total += from(line);
});
await new Promise(res => reader.once("close", res));
const snafu = to(total);
console.log("SNAFU number to supply to Bob's console:", snafu);
